src/api/spec/models/workflow_spec.rb

```diff
@@ -42,11 +45,37 @@
         }
       end
       let!(:target_project) { create(:project, name: 'test-target-project:openSUSE:open-build-service:PR-4', maintainer: user) }
+      let!(:target_package) { create(:package, name: 'test-target-package', project: target_project) }
+
+      context 'we are dealing with a branch package step' do
+        before { login user }
+
+        it 'removes the target project' do
+          expect { subject.call }.to change(Project, :count).from(2).to(1)
+        end
+
+        it 'deletes event subscriptions' do
+          EventSubscription.create!(channel: 'scm', token: token, receiver_role: 'maintainer', eventtype: 'Event::BuildFail', package: target_package)
+          expect { subject.call }.to change(EventSubscription, :count).from(1).to(0)
+        end
+      end
+
+      context 'when we are dealing with a configure project step' do
+        let(:yaml) do
+          { 'steps' => [
+            {
+              'configure_repositories' => {
+                'project' => 'test-target-project'
+              }
+            }
+          ] }
+        end
 
-      before { login user }
+        before { login user }
 
-      it 'removes the target project' do
-        expect { subject.call }.to change(Project, :count).from(2).to(1)
+        it 'does not remove the target project' do
```

Comment:
Could we also check that no event subscriptions are deleted?

Reply:
I was unsure if to add this test or not... I'll add it

Reply:
Added